### ISSUE TITLE
Installation issue: libunwind

### ISSUE THREAD

**Issue by BenWibking** [LLNL/spack#46024]
### Steps to reproduce the issue

```console
$ spack spec -I libunwind@1.6.2%gcc@14.2.1

Input spec
--------------------------------
 -   libunwind@1.6.2%gcc@14.2.1

Concretized
--------------------------------
 -   libunwind@1.6.2%gcc@14.2.1~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-fedora_asahi_remix40-m1
[+]      ^gcc-runtime@14.2.1%gcc@14.2.1 build_system=generic arch=linux-fedora_asahi_remix40-m1
[e]      ^glibc@2.39%gcc@14.2.1 build_system=autotools arch=linux-fedora_asahi_remix40-m1
[+]      ^gmake@4.4.1%gcc@14.2.1~guile build_system=generic arch=linux-fedora_asahi_remix40-m1
[+]      ^gnuconfig@2022-09-17%gcc@14.2.1 build_system=generic arch=linux-fedora_asahi_remix40-m1
```

### Error message

<details><summary>Error message</summary>

<pre>
   382    libtool: compile:  /home/benwibking/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I../include -I../in
            clude -I../include/tdep-aarch64 -I. -D_GNU_SOURCE -DNDEBUG -g -O2 -fexceptions -Wall -Wsign-compare -D__
            EXTENSIONS__ -MT aarch64/Ginit_remote.lo -MD -MP -MF aarch64/.deps/Ginit_remote.Tpo -c aarch64/Ginit_rem
            ote.c  -fPIC -DPIC -o aarch64/.libs/Ginit_remote.o
     383    aarch64/Ginit_local.c: In function 'unw_init_local_common':
  >> 384    aarch64/Ginit_local.c:51:9: error: assignment to 'ucontext_t *' from incompatible pointer type 'unw_cont
            ext_t *' {aka 'unw_tdep_context_t *'} [-Wincompatible-pointer-types]
     385       51 |   c->uc = uc;
     386          |         ^
  >> 387    make[2]: *** [Makefile:4115: aarch64/Ginit_local.lo] Error 1
</pre></details>

### Information on your system

* **Spack:** 0.23.0.dev0 (94c99fc5d44a7e0a2c81fbfe8ca08356bde7a5c7)
* **Python:** 3.12.4
* **Platform:** linux-fedora_asahi_remix40-m1

### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/16741681/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/16741682/spack-build-env.txt)

@mwkrentel

Is this just a gcc 14 incompatibility? If so, maybe gcc-14 should be blocked in the package recipe.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by mwkrentel** [comment]
@BenWibking   Exactly what hardware, OS and/or container are you running?
It looks like Linux on Macbook with M1, is that right?

I don't have a good way to replicate that.

It looks like the problem is an aarch64 issue with gcc 14.
My access to aarch64 is a bit hampered right now.
Anyway, it compiles fine with gcc 14 on x86_64.

So, if you want, you could write a PR to add a conflict for gcc 14,
but restrict it to when(aarch64).

Also, file an issue with libunwind on github, let's see what they say.
I took a brief look, but I didn't see anything related.

**Comment by BenWibking** [comment]
Yes, it's Linux on a M1 Macbook. I'll open an issue with libunwind.

**Comment by mwkrentel** [comment]
Yes, if libunwind can provide a short patch to fix the build,
that would be better than a conflict.
